app/org/sagebionetworks/bridge/services/UserAdminService.java

```
@@ -117,50 +118,51 @@ public UserSession createUser(Study study, StudyParticipant participant, Subpopu
                 ConsentSignature signature = new ConsentSignature.Builder().withName(name)
                         .withBirthdate("1989-08-19").withSignedOn(DateUtils.getCurrentMillisFromEpoch()).build();
                 
                 if (subpopGuid != null) {
-                    consentService.consentToResearch(study, subpopGuid, newUserSession.getParticipant(), signature, NO_SHARING, false);
+                    consentService.consentToResearch(study, subpopGuid, updatedParticipant, signature, NO_SHARING, false);
                 } else {
-                    for (ConsentStatus consentStatus : newUserSession.getConsentStatuses().values()) {
+                    Map<SubpopulationGuid,ConsentStatus> statuses = consentService.getConsentStatuses(context);
+                    for (ConsentStatus consentStatus : statuses.values()) {
                         if (consentStatus.isRequired()) {
                             SubpopulationGuid guid = SubpopulationGuid.create(consentStatus.getSubpopulationGuid());
-                            consentService.consentToResearch(study, guid, newUserSession.getParticipant(), signature, NO_SHARING, false);
+                            consentService.consentToResearch(study, guid, updatedParticipant, signature, NO_SHARING, false);
                         }
                     }
                 }
-                newUserSession = authenticationService.getSession(study, context);
             }
-            if (!signUserIn) {
-                authenticationService.signOut(newUserSession);
-                newUserSession.setAuthenticated(false);
+            if (signUserIn) {
+                SignIn signIn = new SignIn.Builder().withStudy(study.getIdentifier()).withEmail(participant.getEmail())
+                        .withPassword(participant.getPassword()).build();
+                return authenticationService.signIn(study, context, signIn);
             }
```

This only works if the user is created with an email. Without it, it'll throw 400 InvalidEntityException.

Do we know for sure this is only called for users with emails? Should we have a check for if participant.getEmail() exists and throw a different exception? Fail silently?

---

No, we don't, can also be called with an account using only phone. Changed and verified with test.